src/RapideFix/Validation/FixVersionValidator.cs

```diff
@@ -0,0 +1,37 @@
+﻿using RapideFix.DataTypes;
+using System;
+
+namespace RapideFix.Validation
+{
+  public class FixVersionValidator : IValidator
+  {
+    public bool IsValid(Span<byte> data, FixMessageContext messageContext)
+    {
+      if(!data.StartsWith(KnownFixTags.FixVersion))
+      {
+        return false;
+      }
+      int beginStringLength = data.IndexOf(Constants.SOHByte);
+      var fixVersionValue = data.Slice(KnownFixTags.FixVersion.Length, beginStringLength - KnownFixTags.FixVersion.Length);
+      if(!SupportedFixVersion.TryParse(fixVersionValue, out var version))
+      {
+        return false;
+      }
+      messageContext.FixVersion = version;
+      var slice = data.Slice(beginStringLength);
+      bool isValid = true;
+      if(version == SupportedFixVersion.Fix50)
+      {
+        isValid &= slice.IndexOf(KnownFixTags.SenderCompId) >= 0
+          && slice.IndexOf(KnownFixTags.TargetCompId) >= 0;
+      }
+      var msgTypeIndex = slice.IndexOf(KnownFixTags.MessageType);
+      var lengthIndex = slice.IndexOf(KnownFixTags.Length);
+      messageContext.MessageTypeTagStartIndex = messageContext.MessageTypeTagStartIndex;
```

Comment:
I think you wanted to set the value of `msgTypeIndex `here.

Reply:
Hmmm, or I did not want to set it at all. Need to double check.